wp-web-push/lib/delapuente/wp-sw-concoct/class-wp-sw-concoct-router.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+class WP_SW_Concoct_Router {
+    private static $TRIGGER = '_wpswconcoct';
+
+    private static $instance;
+
+    // TODO: Perhaps it's better to separate init() from getting the singleton
+    public static function get_router() {
+        if (!self::$instance) {
+            self::$instance = new self();
+        }
+        return self::$instance;
+    }
+
+    private $routes;
+
+    public function add_route($url, callable $callback) {
+        $components = parse_url($url);
+        $path = $components['path'];
+        $this->routes[$path] = $callback;
+    }
+
+    public function route($url) {
```

Comment:
Agree but let me think how to do it. I will return the real route URL in `add_route()` but let me see how the API fits better after rewriting the client code to be easy to read and self-explanatory.